app/src/main/java/org/gdg/frisbee/android/pulse/PulseActivity.java

```diff
@@ -164,15 +164,12 @@ private void initSpinner(String selectedPulse) {
         toolbar.addView(spinnerContainer, lp);
 
         mSpinner = (Spinner) spinnerContainer.findViewById(R.id.actionbar_spinner);
-        mSpinner.setAdapter(mSpinnerAdapter);
         mSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
             @Override
             public void onItemSelected(final AdapterView<?> parent, final View view, final int position, final long id) {
                 String previous = mViewPagerAdapter.getSelectedPulseTarget();
-                mViewPagerAdapter.setSelectedPulseTarget(mSpinnerAdapter.getItem(position));
                 if (!previous.equals(mSpinnerAdapter.getItem(position))) {
-                    Timber.d("Switching chapter!");
-                    mViewPagerAdapter.notifyDataSetChanged();
+                    refreshSpinner(mSpinnerAdapter.getItem(position));
```

Comment:
ViewPager's notifyDataSetChanged is not working because it aggressively cache the Fragment state which shouldn't be cached because we are changing the list completely. Eg. first we show countries, then we show cities in a specific country.